### ISSUE TITLE
Document binary_parameters connection parameter

### ISSUE THREAD

**Issue by kevinburkeshyp** [lib/pq#435]
I'm trying to send/read UUID's from the database. My first thought was that UUID's were 16 bytes, so I should be able to send `[]byte` to the database, but this failed with the `22021` error. 

Reading through the tests, it seems like I can only do this if I pass `binary_parameters=yes` as part of the connection string. However, the consequences of doing so are not very clear; it seems like this goes down a branch that disables prepared statements?

It would be nice if there was more documentation/help around how to pass UUID's to the database; I'm just using a string for now.

**Comment by johto** [comment]
> Reading through the tests, it seems like I can only do this if I pass binary_parameters=yes as part of the connection string. However, the consequences of doing so are not very clear; it seems like this goes down a branch that disables prepared statements?

No, it doesn't do that.  It just uses the underlying protocol a bit differently.

But yeah, with that option byte slices are passed to the database as-is in binary, which means that passing a UUID as a 16-byte slice works directly.

> It would be nice if there was more documentation/help around how to pass UUID's to the database; I'm just using a string for now.

Yup, that's how you do it.  Every type not recognized by database/sql has to be passed as the text representation of the type, except for bytea values which are converted in the driver automatically for historical reasons.  We can't possibly be expected to list how to pass every single type in the documentation, but we could probably do a bit better.

**Comment by cbandy** [comment]
Here's some relevant PostgreSQL documentation: http://www.postgresql.org/docs/current/static/protocol-overview.html#PROTOCOL-FORMAT-CODES

The binary format of nontrivial types is ... deep knowledge.

**Comment by arp242** [comment]
pq should "just do the right thing" instead of having a connection parameter. I'm fairly sure we can do that.

So in short, I want to move away from `binary_parameters` doing anything, and I don't think this needs to be documented.
